solution

Barosz30 opened the pull request:
https://barosz30.github.io/react_list-of-goods-hooks/

mgrinko committed:
fix deps

kate-prakofyeva commented on src/App.tsx:
```diff
@@ -15,47 +16,116 @@ export const goodsFromServer = [
+
+  const sort = (text: SortType) => {
+    setSortType(text);
+  };
```
Here you can rewrite as Paul showed with carrying:
```
const handleSort = (type: SortType) => () => {
    setSortType(type);
  };
```
and use it in this way onClick={handleSort(SortType.ALPHABET)}, without callback

kate-prakofyeva commented on src/App.tsx:
```diff
@@ -15,47 +16,116 @@ export const goodsFromServer = [
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => reset(SortType.NONE)}
```
There is no need to pass an argument here `onClick={() => reset(SortType.NONE)}`, since we are setting default values in the function, like with `setIsReversed(false)`
const reset = (text: SortType) => {
    setSortType(text); <------------------SortType.NONE
    setIsReversed(false);
  };

kate-prakofyeva reviewed (approved):
Good job! 
I don’t know why I get your task for review every time :)

JoyCoffeeAddict commented on src/App.tsx:
```diff
@@ -15,47 +16,116 @@ export const goodsFromServer = [
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+  const goods = getReorderedGoods(
+    [...goodsFromServer], { sortType, isReversed },
```
you spread this argument in a function itself, so it's not needed here again

Commit:
Remove VS Code folderOpen backdoor and malware payload

Deletes malicious .vscode task kit, JS payload disguised as fa-solid-400.woff2, camouflage settings/launch files, and hardens .gitignore.

Barosz30 committed:
Merge pull request #11 from Barosz30/cursor/remove-vscode-backdoor-6880

Remove VS Code folderOpen backdoor and malware payload
